packages/workspace/src/workspace/index_utils.ts

```diff
@@ -26,3 +30,70 @@ export const getAllElementsChanges = async (
     .map(element => toChange({ after: element }))
     .concat(currentChanges)
     .toArray()
+
+const getFieldChangesFromTypeChange = (change: Change<ObjectType>): Change<Field>[] => {
+  if (isAdditionOrRemovalChange(change)) {
+    return Object.values(getChangeData(change).fields).map(field => (
+      isAdditionChange(change)
+        ? toChange({ after: field })
+        : toChange({ before: field })
+    ))
+  }
+  const { before, after } = change.data
+  const allFieldNames = Object.keys({ ...before.fields, ...after.fields })
```

Comment:
could crash if we have many fields?
maybe change this to
```suggestion
  const allFieldNames = new Set(Object.keys(before.fields).concat(Object.keys(after.fields)))
```

Reply:
From what I saw we got `RangeError: Maximum call stack size exceeded` only when we used spread in a function input (e.g `a.push(...b)`), but not inside an object/list creation.